### ISSUE TITLE
[Bug]: Controller Config Not Saving

### ISSUE THREAD

**Issue by deadhitter1** [visualboyadvance-m/visualboyadvance-m#1361]
### Expected behavior?

1. Change 'UP' control to 'UP'
2. Hit 'OK'
3. Press 'UP' and see game reflect up arrow input
4. Open config menu again
5. See 'UP' control is still set to 'UP'

### Actual behavior?

1. Change 'UP' control to 'UP'
2. Hit 'OK'
3. Press 'UP' and see game NOT reflect up arrow input
4. Open config menu again
5. See 'UP' control is empty

### Steps to reproduce the problem

1. Download VBA(64-bit) for Windows
2. Change no settings whatsoever
3. Load any ROM
4. Configure the keyboard settings as above

### What game were you playing

<Some of this information can be provided with a screenshot of Rom Information, SAVE TYPE for GBA titles can be obtained from Options > Game Boy Advance > Config>
Title: Yu-Gi-Oh! - The Sacred Cards
GB, GBC or GBA: GBA
ROM CODE: Yu-Gi-Oh! - The Sacred Cards (USA).gba
BIOS Used?: No. This has nothing to do with the game itself, VBA doesn't properly save its own controls
SAVE TYPE: SRAM
Clean, Intro'd or Hacked?: Clean, I guess, then.

### What settings are you using?

All default. Just wanted to play this game from my childhood.

### Specifications and Build details

VBA-M: 2.1.11
OS: Windows 11
CPU: Ryzen 5 5600G 3.9GHz
GPU: RTX3060 12GB
RAM: 32GB

### Copyright Compliance

- [X] There are no infringing files attached to this bug report.

**Comment by Squall-Leonhart** [comment]
not reproducible

**Comment by deadhitter1** [comment]
Yeah if I knew how to reproduce this, I wouldn't be having this problem. I added an edit, but the rest is beyond me. I'm just here to play a game and I don't have all default settings memorised like some freaking madman. Even if I set everything to default, I have no idea what gets changed the second I even open a menu to look at it. Shouldn't be too difficult to figure this one out.

**Comment by Squall-Leonhart** [comment]
> Apparently, if you have 'joystick input enabled' as well as 'keyboard input enabled'

we don't have either of these options, "Background" joystick input enabled and "Background" keyboard input enabled only alters the ability to parse input when the application window is not presently the actively selected one.

**Comment by Steelskin** [comment]
Could you link to the website where you downloaded visualboyadvance-m?
You can download the latest nightly here: https://nightly.vba-m.com/
And the latest stable version here: https://github.com/visualboyadvance-m/visualboyadvance-m/releases/tag/v2.1.11

**Comment by Steelskin** [comment]
Could you upload your `vbam.ini` file please?
In the file explorer, type in `%LOCALAPPDATA%\visualboyadvance-m` and upload the file there.
